### PR TITLE
Add coverage target

### PR DESCRIPTION
**Pull request by BassemMohsen**
* Update CMakeLists to configure the build to generate unit testing coverage
results
* Add build target for coverage results

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,6 +14,13 @@ find_package(PkgConfig)
 # Define GNU standard installation directories
 include(GNUInstallDirs)
 
+# Generate code coverage counters and HTML report
+if ( CMAKE_BUILD_TYPE STREQUAL "Coverage" OR CMAKE_BUILD_TYPE STREQUAL "Debug" )
```
Comment (GordanM) [inline CMakeLists.txt]: Sometimes you leave a space between the `( CMAKE` and sometimes you don't `(CMAKE_CXX_FLAGS)`. Could you please change the commit so these are consistent?

**Review by GordanM — CHANGES_REQUESTED** [review]
Good stuff, some minor comment and the signed-off tag is missing.

File: README.md [hunk 2]
```diff
@@ -3,6 +3,22 @@ TEMPLATE-LIBRARY
 A template C/C++ library. Uses CMake as build system and
 [googletest](https://github.com/google/googletest) as testing framework.
 
+## Coverage results
+This project uses tools such as lcov, gcov, genhtml to collect and generate code
```
Comment (kursatkobya) [inline README.md]: could you remove words "such as"

**Review by kursatkobya — CHANGES_REQUESTED** [review]
just a minor comment

**Review by kursatkobya — APPROVED** [review]
Looks good to me.

**Comment by GordanM** [comment]
@BassemMohsen usually you shouldn't merge a commit that has a red `X` but since you've fixed all the issues I presume it's fine this time.

**Pull request merged**